src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.ShoppingCartDao;
+import com.cinema.exceptions.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.saveOrUpdate(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert to DB ShoppingCart: "
+                    + shoppingCart.toString(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "left join fetch sc.user"
```

Comment:
we need left join for tickets, but not for user, for it's more logical to user inner

Reply:
done